project/preprocess_dockerfile.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+echo "#!/bin/bash"
+echo "cat << EOF"
+regex='^(if|elif|else|fi).*$'
+regex2='.*\\$'
+regex3='(.*)(\$|`)(.*)'
+while IFS='' read -r line
+do
+    if [[ $line =~ $regex ]]; then
+        echo "EOF"
+        echo "$line" 
+	echo "cat << EOF"
+    else
+	if [[ $line =~ $regex2 ]]; then
+		line="$line\\"
+	fi    
+	if [[ $line =~ $regex3 ]]; then
+        	templine="${BASH_REMATCH[1]}"
+		result="\\${BASH_REMATCH[2]}${BASH_REMATCH[3]}"
+		while [[ $templine =~ $regex3 ]]; do
+			templine="${BASH_REMATCH[1]}"
+			result="\\${BASH_REMATCH[2]}${BASH_REMATCH[3]}$result"
+		done
+		line="$templine$result"
+    	fi
+        echo "$line"
+    fi  
+done <$(dirname "$BASH_SOURCE")/../Dockerfile.in
+echo "EOF"
+
```

Comment:
The script got complicated only because there are "$", "\" and "`" characters in the Dockerfile that bash tries to process.